File: app/src/main/java/nl/rsdt/japp/jotial/data/structures/area348/AutoEigenaarInfo.java
package nl.rsdt.japp.jotial.data.structures.area348;

import android.os.Parcel;
import android.os.Parcelable;

public class AutoEigenaarInfo implements Parcelable {
    public String autoEigenaar;
    protected AutoEigenaarInfo(Parcel in) {
        autoEigenaar = in.readString();
    }
    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int fags) {
        dest.writeString(autoEigenaar);
    }
    public static final Creator<AutoEigenaarInfo> CREATOR = new Creator<AutoEigenaarInfo>() {
        @Override
        public AutoEigenaarInfo createFromParcel(Parcel in) {
            return new AutoEigenaarInfo(in);
        }

        @Override
        public AutoEigenaarInfo[] newArray(int size) {
            return new AutoEigenaarInfo[size];
        }
    };
}

File: app/src/main/java/nl/rsdt/japp/jotial/data/structures/area348/AutoEigenaarInfo.kt
package nl.rsdt.japp.jotial.data.structures.area348

import android.annotation.SuppressLint
import android.os.Parcel
import android.os.Parcelable

class AutoEigenaarInfo protected constructor(`in`: Parcel) : Parcelable {
    var autoEigenaar: String? = null

    init {
        autoEigenaar = `in`.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, fags: Int) {
        dest.writeString(autoEigenaar)
    }

    companion object {
        @SuppressLint("ParcelCreator")
        val CREATOR: Parcelable.Creator<AutoEigenaarInfo> = object : Parcelable.Creator<AutoEigenaarInfo> {
            override fun createFromParcel(`in`: Parcel): AutoEigenaarInfo {
                return AutoEigenaarInfo(`in`)
            }

            override fun newArray(size: Int): Array<AutoEigenaarInfo?> {
                return arrayOfNulls(size)
            }
        }
    }
}
